src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -916,10 +917,35 @@ private void OutputListToOutputWindow(string prefix, IEnumerable<string> values)
                     continue;
                 }
 
-                if (addedReferencePaths.Add(path))
+                if (metadataReferencesToAdd.TryGetValue(path, out var existingProperties))
                 {
-                    projectContext.AddMetadataReference(path, reference.Properties);
+                    // Merge existing aliases the properties that are already there
+                    var allAliases = existingProperties.Aliases.AddRange(reference.Properties.Aliases);
+
+                    // If one is empty and the other isn't, then we need to toss the global alias in too. The other cases
```

Comment:
Sheesh! Note that this is total curiosity on my part and has no bearing on your PR, but why in the world did we make this API so easy to screw up? It seems like "global" should always be implicit.